src/components/SnapshotImage.tsx

```diff
@@ -106,47 +141,87 @@ export const SnapshotImage = ({
   const showDiff = hasDiff && diffVisible;
   const showFocus = hasFocus && focusVisible;
 
+  const [latestImageLoaded, setLatestImageLoaded] = React.useState(false);
+  const [baselineImageLoaded, setBaselineImageLoaded] = React.useState(false);
+  const [comparisonImageLoaded, setComparisonImageLoaded] = React.useState(false);
+  const [focusImageLoaded, setFocusImageLoaded] = React.useState(false);
+  const snapshotImageLoaded = baselineImageVisible ? baselineImageLoaded : latestImageLoaded;
+
+  let overlayImageLoaded = true;
+  if (showDiff && showFocus) {
+    overlayImageLoaded = comparisonImageLoaded && focusImageLoaded;
+  } else if (showDiff) {
+    overlayImageLoaded = comparisonImageLoaded;
+  } else if (showFocus) {
+    overlayImageLoaded = focusImageLoaded;
+  }
+  const loading = !latestImageLoaded || !overlayImageLoaded;
+
   return (
-    <Container {...props} {...containerProps}>
+    <Container {...props} {...containerProps} loading={loading && !hasError}>
       {latestImage && (
-        <img
-          alt={`Latest snapshot for the '${storyName}' story of the '${componentName}' component`}
-          src={latestImage.imageUrl}
+        <div
           style={{
-            display: baselineImageVisible ? "none" : "block",
+            position: baselineImageVisible ? "absolute" : "static",
+            visibility: baselineImageVisible ? "hidden" : "visible",
+            aspectRatio: `${latestImage.imageWidth} / ${latestImage.imageHeight}`,
+            width: latestImage.imageWidth,
+            maxWidth: "100%",
+            minHeight: 100,
```

Comment:
I prefer to reduce the JSX syntax to a minimum and outsource these inline stylings into a dedicated styled component so that the return part of this component is readable.